### PR TITLE
Support -o rhv-upload option for virt-v2v

### PR DESCRIPTION
**Pull request by xiaodwan**
Add -o rhv-upload option for convert_vm_to_ovirt file.

Signed-off-by: xiaodwan <xiaodwan@redhat.com>

### CODE REVIEW

**Comment by xiaodwan** [comment]
# https://github.com/avocado-framework/avocado-vt/pull/1683

**Comment by xiaodwan** [comment]
@chloerh Please help to review it. Thanks!

File: v2v/tests/cfg/convert_vm_to_ovirt.cfg [hunk 1]
```diff
@@ -16,6 +16,26 @@
     status_test_command = "echo $?"
     v2v_timeout = '1200'
 
+    variants:
+        - rhv_upload:
+            output_method = "rhv_upload"
+            rhv_upload_passwd = ${ovirt_engine_password}
+            rhv_upload_passwd_file = "/tmp/rhv_upload_passwd_file"
```
Comment (chloerh) [inline v2v/tests/cfg/convert_vm_to_ovirt.cfg]: Should be a string to be replaced

**Comment by chloerh** [comment]
Please paste test result of the new cases.

File: v2v/tests/cfg/convert_vm_to_ovirt.cfg (hunk 1 above)
Comment (xiaodwan) [reply v2v/tests/cfg/convert_vm_to_ovirt.cfg]: It's a destination file.

**Comment by xiaodwan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.kvm.linux.7_5.arch_x86_64.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer: PASS (880.09 s)

**Comment by xiaodwan** [comment]
(1/3) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.kvm.linux.7_5.arch_x86_64.qcow2_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer: PASS (766.54 s)
 (2/3) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.kvm.linux.7_5.arch_x86_64.raw_f.NFS.rhv: PASS (494.07 s)
 (3/3) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.kvm.linux.7_5.arch_x86_64.qcow2_f.NFS.rhv: PASS (510.26 s)

**Comment by xiaodwan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.kvm.linux.7_5.arch_x86_64.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (885.81 s)

**Comment by xiaodwan** [comment]
(1/1) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.kvm.linux.7_5.arch_x86_64.raw_f.FC.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (744.10 s)

**Comment by xiaodwan** [comment]
@chloerh @chunfuwen please help to review it, thanks!

**Comment by xiaodwan** [comment]
(1/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.xen.vm.hvm.linux.5_11.arch_i386.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer.nopreallocated: PASS (1088.01 s)
 (2/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.xen.vm.hvm.linux.5_11.arch_i386.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (870.92 s)
 (3/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.xen.vm.hvm.linux.5_11.arch_i386.raw_f.FC.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (864.66 s)
 (4/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.xen.vm.hvm.windows.win2008r2.arch_x86_64.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer.nopreallocated: PASS (1168.34 s)
 (5/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.xen.vm.hvm.windows.win2008r2.arch_x86_64.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: FAIL (1001.12 s)
 (6/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.7_5.arch_x86_64.raw_f.NFS.rhv_upload.rhv_direct.rhv_verifypeer.nopreallocated: PASS (2134.29 s)
 (7/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.7_5.arch_x86_64.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (2424.10 s)
 (8/8) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.esx.vm.6_7.linux.7_5.arch_x86_64.raw_f.FC.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (2402.01 s)

**Comment by xiaodwan** [comment]
Above failed case is due to environment issue, it passed after rerun it.
(1/1) type_specific.io-github-autotest-libvirt.convert_vm_to_ovirt.xen.vm.hvm.windows.win2008r2.arch_x86_64.raw_f.ISCSI.rhv_upload.rhv_direct.rhv_verifypeer.preallocated: PASS (1110.03 s)

File: v2v/tests/src/convert_vm_to_ovirt.py [hunk 2]
```diff
@@ -39,6 +41,14 @@ def run(test, params, env):
     address_cache = env.get('address_cache')
     v2v_opts = params.get("v2v_opts")
     v2v_timeout = int(params.get('v2v_timeout', 1200))
+    output_method = params.get("output_method")
```
Comment (chunfuwen) [inline v2v/tests/src/convert_vm_to_ovirt.py]: add one comment here indicate what those parameter are for?

File: v2v/tests/src/convert_vm_to_ovirt.py [hunk 3]
```diff
@@ -171,3 +195,6 @@ def run(test, params, env):
             os.remove(vpx_passwd_file)
         if hypervisor == "xen":
             process.run("ssh-agent -k")
+        if output_method == 'rhv_upload':
+            os.remove(rhv_passwd_file)
```
Comment (chunfuwen) [inline v2v/tests/src/convert_vm_to_ovirt.py]: better do judgement whether file exist or not before remove them

File: v2v/tests/src/convert_vm_to_ovirt.py (hunk 2 above)
Comment (xiaodwan) [reply v2v/tests/src/convert_vm_to_ovirt.py]: Done

**Pull request merged**